### ISSUE TITLE
Make device uniquely identifiable via the API

### ISSUE THREAD

**Issue by frenck** [itead/Sonoff_Devices_DIY_Tools#58]
Currently, the `deviceid` is only published via mDNS, there is no way to get it via the API. Hence, without mDNS, there is no way to uniquely identify the device.

This causes problems for platforms, like Home Assistant, to manage multiple devices on a single network. A device has to be uniquely identifiable on the network, even without mDNS.

Suggestion: Add `deviceid` to the `zeroconf/info` response.

**Comment by Daniel-zhan-itead** [comment]
A device has to be uniquely identifiable on the network, you have some ways to get it

1. the hostname
2. the id in txtrecord
3. IP address

As sonoff device uses the mDNS protocol to discover the device, without the mDNS, how do you find the device in the network?

**Comment by frenck** [comment]
hostname & IP addresses are not properties of the device and therefore not a unique identifier of the device. For example, I can log in to my router, and change the IP address & hostname via the DHCP configuration. In conclusion, it is not a device identification, but a network identification of the device, which is not what I requested.

The ID is the TXT record is only available via mDNS, which is not a reliable source for home automation in general, for several reasons:

- mDNS does not have to be supported by the client network.
- mDNS with home automation client in Docker may have issues, since mDNS on Docker is not supported (multicast issues).

In conclusion of the above, hostname & IP are not device identifiers but network properties. The device ID is currently only available via mDNS, which is not guaranteed to be able.

Hence, I'm looking for a reliable fallback method to obtain a unique device ID and thus request it to add a method to obtain it via the HTTP API.

**Comment by Daniel-zhan-itead** [comment]
Hi Frenck,

the New DIY Mode has been released, the deviceid has been added in zeroconf/info.

Daniel

**Comment by frenck** [comment]
Great, that close this issue 👍
